woocommerce/woocommerce
# Add HOC to hide inventory collapsible block when nothing is inside it

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Part of https://github.com/woocommerce/woocommerce-min-max-quantities/issues/283

This is needed to fulfill the following AC:

> When it's the only field under the Accordion (when inventory tracking is disabled), the accordion shouldn't render empty

TODO: The collapsible block file is probably not a good place to add this filter. I wonder what is the best place for it. It is a hook, not a React hook, but a WordPress hook. Maybe we have to create a separate folder for these "general" hooks which are not bound to any specific block.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add the following snippet as a mu-plugin or similar:

```php
<?php

function remove_limit_purchase_block( $block ) {
  $block->remove();
}

add_action( 'woocommerce_block_template_area_product-form_after_add_block_product-limit-purchase', 'remove_limit_purchase_block' );

```
2. Enable the new product editor at /wp-admin/admin.php?page=wc-settings&tab=advanced&section=features
3. Go to Products > Add
4. Go to the Inventory tab
5. Check that the "Advanced" collapsible is not shown
6. Remove the snippet or comment the "remove()" call
7. Refresh the editor
8. Check that it re-appears



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
